module/actuation/FootController/src/FootController.hpp

```diff
@@ -91,6 +142,117 @@ namespace module::actuation {
             ik_task->time = foot_control_task.time;
             ik_task->Htf  = foot_control_task.Htf;
 
+            // Add correction to the torso orientation if enabled
+            if (foot_control_task.correction_enabled) {
+                // Hwt quaternion
+                Eigen::Quaterniond Hwt_quat(sensors.Htw.inverse().linear());
+
+                // Get fused roll and pitch
+                double fused_roll;
+                double fused_pitch;
+                FusedFromQuat(Hwt_quat, fused_pitch, fused_roll);
+
+                if (log_level <= DEBUG) {
+                    // graph the correction being applied
+                    emit(graph("Balance/Actual_Roll", fused_roll));
+                    emit(graph("Balance/Actual_Pitch", fused_pitch));
+                }
+
+                // Get the desired roll and pitch
+                Eigen::Quaterniond Hft_quat(ik_task->Htf.inverse().linear());
+
+                double desired_roll;
+                double desired_pitch;
+                FusedFromQuat(Hft_quat, desired_pitch, desired_roll);
+
+                if (log_level <= DEBUG) {
+                    // Graph desired orientation (before PID correction)
+                    emit(graph("Balance/Desired_Roll_Original", desired_roll));
+                    emit(graph("Balance/Desired_Pitch_Original", desired_pitch));
+                }
+
+                // Compute the error between the desired torso orientation and the actual torso orientation
+                auto roll_error  = desired_roll - fused_roll;
+                auto pitch_error = desired_pitch - fused_pitch;
+
+                // If the error is too large, we are probably falling over and should pause applying control.
+                if (std::abs(roll_error) > cfg.max_roll_error || std::abs(pitch_error) > cfg.max_pitch_error) {
+                    if (log_level <= DEBUG) {
+                        log<DEBUG>(fmt::format(
+                            "Balance correction disabled due to large error: roll_error = {}, pitch_error = {}",
+                            roll_error,
+                            pitch_error));
+                    }
+                    return;  // Do not apply control if the error is too large
+                }
```

Comment:
This check works but it stops the walk entirely :/

Reply:
Sorted with [c8dec0d](https://github.com/NUbots/NUbots/pull/1636/commits/c8dec0d36d183939cb92e08d6e96cee7347ebf62)